src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -473,9 +475,31 @@ public void download() {
                     Localization.lang("Fulltext for") + ": " + entry.getCitationKey().orElse(Localization.lang("New entry")));
             downloadTask.showToUser(true);
             downloadTask.onFailure(ex -> {
-                LOGGER.error("Error downloading", ex);
-                dialogService.showErrorDialogAndWait(Localization.lang("Error downloading"), ex);
+            LOGGER.error("Error downloading from URL {}" + urlDownload, ex);
+                if (ex instanceof FetcherClientException) {
+                    FetcherClientException clientException = (FetcherClientException) ex;
+                    int statusCode = clientException.getStatusCode();
+                    if (statusCode == 401 || statusCode == 403) {
+                        dialogService.showErrorDialogAndWait(Localization.lang("Error downloading from URL"),
+                                "HTTP Error " + statusCode + ": Forbidden. This may be due to authentication issues or the server blocking your request. Please check your credentials or contact the server administrator.\nURL: " + urlDownload + "\nPlease check the URL and try again.");
+                    } else if (statusCode == 404) {
+                        dialogService.showErrorDialogAndWait(Localization.lang("Error downloading from URL"),
+                                "HTTP Error " + statusCode + ": Not Found. The requested resource could not be found.\nURL: " + urlDownload + "\nPlease check the URL and try again.");
+                    } else {
+                        dialogService.showErrorDialogAndWait(Localization.lang("Error downloading from URL"),
+                                "HTTP Error " + statusCode + ": " + clientException.getMessage() + "\nURL: " + urlDownload + "\nPlease check the URL and try again.");
+                    }
+                } else if (ex instanceof FetcherServerException) {
+                    FetcherServerException serverException = (FetcherServerException) ex;
+                    int statusCode = serverException.getStatusCode();
+                    dialogService.showErrorDialogAndWait(Localization.lang("Error downloading from URL"),
```

Comment:
The second part is also shown to the user
. Thus, that needs to be localized too. It is OK for me to include the server error here.